packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -144,10 +145,11 @@ export default createRule<[], MessageId>({
                 ]),
               ];
 
-              const stringValue = getStaticStringValue(expression);
-
-              if (stringValue != null) {
-                const escapedValue = stringValue.replace(/([`$\\])/g, '\\$1');
+              if (isLiteral(expression)) {
+                const escapedValue =
```

Comment:
I think you may want to do the escaping in both branches of the ternary, since non-strings may still stringify to something with characters needing escapes. Consider these sneaky test cases:
```typescript
`the nonsense regex ${/${}/} matches the end of a line followed by a pair of braces`;
`the regex ${/\${}/} matches a dollar sign followed by a pair of braces`;
```

Reply:
What a nice catch! I overlooked the fact that we can stringify regexes... Thanks!